database/mssql/scripts/dbscripts/PSP_PIMS_S25_00/Alter Down/01_PSP_PIMS_S25_00_Alter_Down.sql

```diff
@@ -0,0 +1,663 @@
+	DROP CONSTRAINT [PIMS_RSFLPR_H_PK]
+GO
+IF @@ERROR <> 0 SET NOEXEC ON
+GO
+
+-- Drop primary key constraint dbo.RSFLPR_PK
+PRINT N'Drop primary key constraint dbo.RSFLPR_PK'
+GO
+ALTER TABLE [dbo].[PIMS_RESEARCH_FILE_PURPOSE]
+	DROP CONSTRAINT [RSFLPR_PK]
+GO
+IF @@ERROR <> 0 SET NOEXEC ON
+GO
+
+-- Drop table dbo.PIMS_RESEARCH_FILE_PURPOSE
+PRINT N'Drop table dbo.PIMS_RESEARCH_FILE_PURPOSE'
+GO
+DROP TABLE [dbo].[PIMS_RESEARCH_FILE_PURPOSE]
+GO
+IF @@ERROR <> 0 SET NOEXEC ON
+GO
+
+-- Drop primary key constraint dbo.RSHPRT_PK
+PRINT N'Drop primary key constraint dbo.RSHPRT_PK'
+GO
+ALTER TABLE [dbo].[PIMS_RESEARCH_PURPOSE_TYPE]
+	DROP CONSTRAINT [RSHPRT_PK]
+GO
+IF @@ERROR <> 0 SET NOEXEC ON
+GO
+
+-- Drop view dbo.PIMS_CONTACT_MGR_VW
+PRINT N'Drop view dbo.PIMS_CONTACT_MGR_VW'
+GO
+DROP VIEW [dbo].[PIMS_CONTACT_MGR_VW]
+GO
+IF @@ERROR <> 0 SET NOEXEC ON
+GO
+
+-- Drop primary key constraint dbo.RQSRCT_PK
+PRINT N'Drop primary key constraint dbo.RQSRCT_PK'
+GO
+ALTER TABLE [dbo].[PIMS_REQUEST_SOURCE_TYPE]
+	DROP CONSTRAINT [RQSRCT_PK]
+GO
+IF @@ERROR <> 0 SET NOEXEC ON
+GO
+
+-- Drop sequence dbo.PIMS_RESEARCH_FILE_PURPOSE_ID_SEQ
+PRINT N'Drop sequence dbo.PIMS_RESEARCH_FILE_PURPOSE_ID_SEQ'
+GO
+DROP SEQUENCE [dbo].[PIMS_RESEARCH_FILE_PURPOSE_ID_SEQ]
+GO
+IF @@ERROR <> 0 SET NOEXEC ON
+GO
+
+-- Alter table dbo.PIMS_RESEARCH_FILE
+PRINT N'Alter table dbo.PIMS_RESEARCH_FILE'
+GO
+ALTER TABLE [dbo].[PIMS_RESEARCH_FILE]
+	DROP COLUMN [REQUEST_SOURCE_TYPE_CODE], [REQUESTOR_NAME], [REQUESTOR_ORGANIZATION], [REQUEST_DATE], [ROAD_NAME], [ROAD_ALIAS], [REQUEST_DESCRIPTION], [REQUEST_SOURCE_DESCRIPTION], [RESEARCH_RESULT], [IS_EXPROPRIATION], [EXPROPRIATION_NOTES], [RESEARCH_COMPLETION_DATE]
+GO
+IF @@ERROR <> 0 SET NOEXEC ON
+GO
+
+-- Alter table dbo.PIMS_PROPERTY_RESEARCH_FILE
+PRINT N'Alter table dbo.PIMS_PROPERTY_RESEARCH_FILE'
+GO
+ALTER TABLE [dbo].[PIMS_PROPERTY_RESEARCH_FILE]
+	DROP COLUMN [PROPERTY_NAME]
+GO
+IF @@ERROR <> 0 SET NOEXEC ON
+GO
+
+-- Alter table dbo.PIMS_PROPERTY_RESEARCH_FILE_HIST
+PRINT N'Alter table dbo.PIMS_PROPERTY_RESEARCH_FILE_HIST'
+GO
+ALTER TABLE [dbo].[PIMS_PROPERTY_RESEARCH_FILE_HIST]
+	DROP COLUMN [PROPERTY_NAME]
+GO
+IF @@ERROR <> 0 SET NOEXEC ON
+GO
+
+-- Alter table dbo.PIMS_RESEARCH_FILE_HIST
+PRINT N'Alter table dbo.PIMS_RESEARCH_FILE_HIST'
+GO
+ALTER TABLE [dbo].[PIMS_RESEARCH_FILE_HIST]
+	DROP COLUMN [REQUEST_SOURCE_TYPE_CODE], [REQUESTOR_NAME], [REQUESTOR_ORGANIZATION], [REQUEST_DATE], [REQUEST_SOURCE_DESCRIPTION], [IS_EXPROPRIATION], [RESEARCH_COMPLETION_DATE]
+GO
+IF @@ERROR <> 0 SET NOEXEC ON
+GO
+
+-- Alter table dbo.PIMS_PROP_PROP_TENURE_TYPE
+PRINT N'Alter table dbo.PIMS_PROP_PROP_TENURE_TYPE'
+GO
+ALTER TABLE [dbo].[PIMS_PROP_PROP_TENURE_TYPE] DROP CONSTRAINT [PRPRTT_PROPERTY_TENURE_TYPE_CODE_DEF]
+GO
+IF @@ERROR <> 0 SET NOEXEC ON
+GO
+ALTER TABLE [dbo].[PIMS_PROP_PROP_TENURE_TYPE] ADD CONSTRAINT "PRPRTT_PROPERTY_TENURE_TYPE_CODE_DEF" DEFAULT 'UNKNOWN' FOR [PROPERTY_TENURE_TYPE_CODE]
+GO
```

Comment:
That's the expected behaviour.  If there was a IF NOT EXISTS I'd be using that.  This way the constraint is properly preserved.

Reply:
Ah, I see. Should executing Alter Up and then Alter Down be equal to not executing anything? In this case the alter down is not removing the constrain that the alter up is adding

Reply:
Yes, because:
1. The introduction of a NOT NULL without a default value can cause issues if introduced into a table that already contains data; and
2. Not assigning a name to the constraint results in a randomly named constraint, with that name differing in each environment it's created in.  This makes it more difficult to locate and alter that constraint because you first have to locate the constraint and determine its name before you can operate on it.

The way I've remedied that here is to introduce it in the Alter Down and preserve it in the Alter Up.